android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt

```diff
@@ -144,18 +145,21 @@ class AppLinksInterceptor(
         return null
     }
 
-    // for app links interceptor any domains such as example.com, www.example.com and m.example.com
-    // are considered as the same domain
-    private fun isSameDomain(url1: String?, url2: String?): Boolean {
+    // Determines if the transition between the two URLs is related.  If the two URLs
+    // are from the same website then the app links interceptor will not try to find an application to open it.
+    @VisibleForTesting
+    internal fun isSameDomain(url1: String?, url2: String?): Boolean {
         return stripCommonSubDomains(url1?.tryGetHostFromUrl()) == stripCommonSubDomains(url2?.tryGetHostFromUrl())
     }
 
+    // Remove subdomains that are ignored when determining if two URLs are from the same website.
     private fun stripCommonSubDomains(url: String?): String? {
         return when {
```

Comment:
Drive by: With this method only removing at max one subdomain it allows for others to still be present.
In a quick search I found for example
> https://www.m.999shopbd.com

which can also be accessed through

> https://999shopbd.com

but `isSameDomain` will return `false` while I would've expected to get `true`.
Should I open a new ticket for this?

Reply:
Yes, please open a new issue for this.  I'm not sure for the app links case we need to make that determination but it will be good to investigate.  Thanks

Reply:
Created https://bugzilla.mozilla.org/show_bug.cgi?id=1818492.